compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -2410,6 +2410,23 @@ impl<'tcx> TyCtxt<'tcx> {
         }
     }
 
+    pub fn get_attrs_by_path(
+        self,
+        did: impl Into<DefId>,
+        attr: Box<[Symbol]>,
```

Comment:
Why does this take a `Box`? 😅 

Please use lifetimes here and make it `&'a [Symbol]` -- the allocation here is really unnecessary.